Add section about SSH signing keys to developer docs. (#1755)
NVIDIA/cccl: CONTRIBUTING.md

   For more details on building and testing, refer to the [Building and Testing](#building-and-testing) section below.

4. **Commit Changes**:

   Setup ssh signing keys (if needed, see next section) and commit:

   ```bash
   git commit -m "Brief description of the change"
   ```

### SSH Signing Keys

> CCCL projects require commits to be signed with a cryptographic key.

Reviewer: comment: This isn't quite true. Signed commits are only necessary for internal contributors who want CI to run automatically and not require `/ok to test`.

Author: Ah, ok, I'd misunderstood that nuance. I'll gladly move it to the ci docs, as long as I have something concise to link (and refer back to myself 😄)

To enable commit signing using your existing ssh key, set the following git options:

```bash
git config --global gpg.format ssh

> git config --global user.signingKey "$(cat ~/.ssh/id_rsa.pub)"

Reviewer 2: Nitpick: I would change that to
```suggestion
git config --global user.signingKey "$(cat Location_Of_Your_Public_Key)"
```
